CalibPPS/ESProducers/plugins/CTPPSCompositeESSource.cc

```diff
@@ -323,7 +323,7 @@ void CTPPSCompositeESSource::buildOptics(const edm::ParameterSet &profile, Profi
 
   for (const auto &pset : ctppsOpticalFunctions.getParameter<std::vector<edm::ParameterSet>>("opticalFunctions")) {
     const double &xangle = pset.getParameter<double>("xangle");
-    const std::string &fileName = pset.getParameter<edm::FileInPath>("fileName").fullPath();
+    const std::string fileName = pset.getParameter<edm::FileInPath>("fileName").fullPath();
```

Comment:
@iarspider , if I am not working, both `xangle` and `fileName` can be dropped and we can pass the return values directly to `fileInfo.push_back({....})`